andremun/pyInstanceSpace
# Read files junheng (#78), merge that branch content to main

* make all scripts classes

* alternative for trace class structure

* allow assignment that is never used for demonstration purposes

* make pytest a dev dependancy

* create InstanceSpace

* start work on calling individual stages, need to think about how we store the output of stages ina way that isn't just chucking everything in model

* remove self typing as unneccesary

* more thoughts on storing data between stages

* finish first pass at class representation of instance space

* add helpful alternate constructors. Ben prefers having these outside the InstanceSpace class instead of having them as static methods on the class

* fix spacing

* fix unused import

* make metadata a dataclass

* add todo for docstrings

* finish the reading CSV

* problemmatic jason reading

* test: add unit test for loading csv and json files

* update docstring

* update docstring

* updated json read

 raise error when json had fields that are not defined in the class

* update minor details in options

* fix some poe errors, improve the json loader

* assign none to any top field

* finalized read json, with new testing cases

* corrected all the errors

* delete the _init_ in scripting, waiting for implementation

* Adopt the content of Option in HEAD

* modified Metadata logic to make it is consistant with mian

* need to finalise from_file in metadata

* Update instance_space.py

* minor correction

* refactor

* recover indents

* Update instance_space.py

* Apply suggestions



* minor correction

* improve based on seggestion

* poe correction

* ignore .idea files

* get rid of .idea files

* think about approach for reading metadata files

* more design with chen

* more design with chen

* more design with chen

* need more tests

* fix. option lader

* fix. option loader

* fix. some poe errors

* Architecture Planning v2 (Previously Options Overhaul) (#80) (#89)

* propose options method

* improve docs for options

* make Metadata consistant

* make Metadata consistant

* Data can be changed

* add function to instantiate Options with mostly default values

* fix poe errors

* fix rename of options

* add data to outs

* remove the output of future stages if a previous stage is ran

* add raises when required options aren't present in cloister

* create a from_options function for prelim options

* unify return values of stages

* fix mypy and pytest clash, 1 error in tests

* fix mypy error

* make Data immutable

* make stage state order of arguments consistant with stage output

* remove file handling from this pr

* remove file handling from this pr

* add defaults from example.m

* name default options module something better

* fix poe errors

* remove logging if options where none, as they can't be none

* clean up with poe check

---------




* merge main

* Reading Options is done

* finalizing and fixing poe errors

* all done for the file reading functions

* minor changed

* fix

* fix. Whitespace

* fixed try catch

* fixed the try catch and read from directory

* update docstring, fixed all the white spaces

* fix white space changing

* fix. instance_space_from_directory

* fix. option.py whitespaces

* fix. whitespaces again

* fix. whitespaces in inst_space.py

* refactor instance_space

* fix. Based on the Ben's suggestion

* remove unused files

* test for the pipeline

* test for pipeline

* fix test

* test

* to see CI behavior

* recover test files

* trying without cloister testing

* trying with the cloister

* testing cases fixes

* fix. metadata

---------
